go/tasks/plugins/array/k8s/launcher.go

```diff
@@ -33,8 +33,8 @@ var arrayJobEnvVars = []corev1.EnvVar{
 	},
 }
 
-func formatSubTaskName(_ context.Context, parentName, suffix string) (subTaskName string) {
-	return utils.ConvertToDNS1123SubdomainCompatibleString(fmt.Sprintf("%v-%v", parentName, suffix))
+func formatSubTaskName(_ context.Context, parentName, indexStr, retryAttemptStr string) (subTaskName string) {
```

Comment:
This name is used as the Pod name, right? Won't this break backward compatibility for in-flight (pun intended) tasks? we might need to use [State Version](https://github.com/flyteorg/flyteplugins/blob/master/go/tasks/pluginmachinery/core/state.go#L15) to differentiate between "before" and "after" upgrade...

Reply:
nit: Looks like we always have the retry attempt as a uint and we convert it to a string, should we move the logic to do the uint -> str here instead?

Reply:
@EngHabu in diving through backwards-compatibility for in-flight (love the pun!) tasks I think this is challenging. The difficulty stems from the transition from reporting and tracking retryable failures in the task execution context to the subtask retryAttempts array. Since the pod name is contingent on the retry and these may be incremented independently there are many situations which cause pod name overlap (same name for different retry attempts) if we are not careful.

I've convinced myself there is no way to transition a running map_task (ie. where at least one pod is currently running) to this proposed approach (ie. subtask retryAttempts array). Therefore, any solution requires a collection of if statements checking the state_version to allow currently running map tasks to continue using podnames based solely on the task execution context retry attempt and allowing retryable failures to fall-through to propeller which can fail the entire map task. I think this introduces significant complexity to the codebase, where this logic is already complex enough.

Is there a simple solution that I'm overthinking?

Reply:
nvm, figured it out.